File: SDK-v5/gdpr-places/app/src/main/java/com/localytics/gdpr/UserLoginTask.java
package com.localytics.gdpr;

import android.os.AsyncTask;

/**
 * Represents an asynchronous login/registration task used to authenticate
 * the user.
 */
public class UserLoginTask extends AsyncTask<Void, Void, AuthenticationResult> {

    private final String mEmail;
    private final String mPassword;
    private final LoginTaskHandler mHandler;

    UserLoginTask(String email, String password, LoginTaskHandler handler) {
        mEmail = email;
        mPassword = password;
        mHandler = handler;
    }

    @Override
    protected AuthenticationResult doInBackground(Void... params) {
        try {
            // Simulate network access.
            Thread.sleep(2000);
        } catch (InterruptedException e) {
            return new AuthenticationResult(false, null, false);
        }

        // A proper backend integration should be connected here that will log the user in
        // and check their customer ID for their data opt out status
        String customerIdFromServer = "customerIdFromServer";
        boolean isUserPrivacyOptedOut = false;

        return new AuthenticationResult(true, customerIdFromServer, isUserPrivacyOptedOut);
    }

    @Override
    protected void onPostExecute(final AuthenticationResult result) {
        mHandler.onPostExecute(result);
    }

    @Override
    protected void onCancelled() {
        mHandler.onCancelled();
    }

    public interface LoginTaskHandler {

        void onPostExecute(AuthenticationResult result);

        void onCancelled();
    }
}

File: SDK-v5/kotlin/gdpr/app/src/main/java/com/localytics/gdpr/UserLoginTask.kt
package com.localytics.gdpr

import android.os.AsyncTask

/**
 * Represents an asynchronous login/registration task used to authenticate
 * the user.
 */
class UserLoginTask internal constructor(private val mEmail: String, private val mPassword: String, private val mHandler: LoginTaskHandler) : AsyncTask<Void, Void, AuthenticationResult>() {

    @Override
    override fun doInBackground(vararg params: Void): AuthenticationResult {
        try {
            // Simulate network access.
            Thread.sleep(2000)
        } catch (e: InterruptedException) {
            return AuthenticationResult(false, null, false)
        }

        // TODO A proper backend integration should be connected here that will log the user in
        // and check their customer ID for their data opt out status
        val customerIdFromServer = "customerIdFromServer"
        val isUserPrivacyOptedOut = false

        return AuthenticationResult(true, customerIdFromServer, isUserPrivacyOptedOut)
    }

    @Override
    override fun onPostExecute(result: AuthenticationResult) {
        mHandler.onPostExecute(result)
    }

    @Override
    override fun onCancelled() {
        mHandler.onCancelled()
    }

    interface LoginTaskHandler {

        fun onPostExecute(result: AuthenticationResult)

        fun onCancelled()
    }
}
